### ISSUE TITLE
Why not use PyTorch instead of NumPy everywhere?

### ISSUE THREAD

**Issue by AhmedThahir** [koaning/scikit-mdn#8]
- Every NumPy function has an equivalent PyTorch function, so this makes usage of NumPy redundant.
- From my understanding, PyTorch tensors are faster than numpy arrays.
- This is not critical, and just a suggestion for documentation purpose.

**Comment by koaning** [comment]
The reason to push it out as Numpy is related to what follows after sklearn. Most of the time, scikit-learn assumes a Numpy stack. So it could be that if we return PyTorch that some system later breaks. 

I am open to having the output be determined by a flag though. This library was made in a bit of haste for a livestream and it is very well possible that there are a bunch of improvements missing.

**Comment by AhmedThahir** [comment]
Can't we use PyTorch's .detach() at the end of every function, to guarantee this?

This will perform a zero-copy conversion of a PyTorch tensor to NumPy array.

**Comment by koaning** [comment]
Don't we already do that? The `forward()` pass does that. 

https://github.com/koaning/scikit-mdn/blob/main/skmdn/__init__.py#L135-L151

**Comment by AhmedThahir** [comment]
Yes, that's right. 

So doing detach will guarantee that output with be in NumPy. Hence why not use pytorch functions as much as possible.

**Comment by koaning** [comment]
Ahhh now I see what you mean. 

If you can demonstrate that it provides a meaningful speedup I'll gladly consider a PR. But yeah, it feels like a low-priority feature on this repo (the repo itself is also not incredibly high priority).
